Source/MQTTnet/Internal/Memory.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Runtime.CompilerServices;
+
+namespace MQTTnet.Internal
+{
+    public static class Memory
```

Comment:
* `internal`? I see a lot of types in the "Internal" namespace are `public`, but for new types it should really be `internal`

* Naming is hard (I'm not better in this), but `Memory` is for me Memory/Span, so maybe `CopyExtension` and make the single method an extension method? Or leave as static method and name the type `CopyHelper`?

Reply:
I prefer to keep most things public to make testing easier (i know that "internal visible to" exists). So public is fine in my opinion. The "Internal" namespace is an indicator that these components do not have a stable API.

Reply:
Regarding the: I also would prefer "MqttMemoryHelper" etc. because "Memory" is already in use by the framework itself.

Reply:
I will merge the PR and do the changes myself so that w new version can be released soon.